partitions.go

```diff
@@ -82,28 +85,53 @@ func (p *partitions) getPartitionSize(partition string) (int64, error) {
 	}
 }
 
-func (p *partitions) getAndSetInactivePartition() (string, error) {
+func (p *partitions) getAndCacheInactivePartition() (string, error) {
+	if p.partitionANumber == "" || p.partitionBNumber == "" {
+		return "", ErrorPartitionNumberNotSet
+	}
+	if p.partitionANumber == p.partitionBNumber {
+		return "", ErrorPartitionNumberSame
+	}
+
 	active, err := p.GetActive()
 	if err != nil {
 		return "", err
 	}
 
-	mountSufix := active[len(active)-1:]
-	mountPrefix := active[:len(active)-1]
-
-	log.Debugf("Setting inactive partition %s [%s][%s]", active, mountSufix, mountPrefix)
+	if len(active) < len(p.partitionANumber) ||
+		len(active) < len(p.partitionBNumber) {
+		return "", InvalidActivePartition
+	}
 
-	switch mountSufix {
-	case "2":
-		p.inactive = mountPrefix + "3"
-		return p.inactive, nil
-	case "3":
-		p.inactive = mountPrefix + "2"
-		return p.inactive, nil
-	default:
+	// Check whether the active partition ends with the number of either the
+	// A or the B partition, and then assign the non-matching one to
+	// inactive.
+	var inactive string
+	var itsA, itsB bool
+	if active[len(active)-len(p.partitionANumber):] == p.partitionANumber {
```

Comment:
Because of other comments in the meta-mender PR, I decided to switch to full device names instead of numbers. Which means we can do direct string comparison, so this whole section is much simpler now.